### PR TITLE
fix: [CDS-83828] Improve error handling for rendering of helm chart

### PR DESCRIPTION
**Pull request by alebuh**
## Describe your changes
Improving error handling in two places:
1. Loading String as Yaml object fails we are adding more information to the exception (String value that can't be loaded as Yaml)
2. If the rendering of the helm chart fails, we should not fail execution. Therefore we are catching the exception, printing the warning, and relaying on helm install/upgrade to fail with an appropriate message

Test:
If rendering Helm Chart fails (for example due to an invalid manifest) we will not fail execution. We will print a warning message.
Screenshot 2023-11-21 at 12 15 33

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55665)

### CODE REVIEW

**Comment by alebuh** [comment]
trigger uts

**Comment by alebuh** [comment]
trigger uts,smartchecks

**Comment by alebuh** [comment]
trigger smartchecks

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by alebuh** [comment]
trigger ut8, gitleaks

**Review by abosii-harness — APPROVED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/manifest/ManifestHelper.java [hunk 1]
```diff
@@ -129,7 +129,7 @@ private static Map readKubernetesSpecAsMap(String spec) {
       if (o instanceof Map) {
         return (Map) o;
       } else {
-        throw new KubernetesYamlException("Invalid Yaml. Object is not a map.");
+        throw new KubernetesYamlException(format("Invalid Yaml. Spec: %n %s can't be loaded as map", spec));
```
[the remark this answers is not in the document]
Comment (abhinavsingh1196) [reply 970-api-services-beans/src/main/java/io/harness/k8s/manifest/ManifestHelper.java:132]: although if its not even a valid map, we should probably be good 🤔

**Review by abhinavsingh1196 — APPROVED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/manifest/ManifestHelper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (alebuh) [reply 970-api-services-beans/src/main/java/io/harness/k8s/manifest/ManifestHelper.java:132]: The exception should be sanitized and that should take care of the secrets, right? 
For this specific case, with printing helm chart, we are not going to return this exception to the UI, we are going to handle it and log it in GCP to help us debug an issue if it appears

**Comment by alebuh** [comment]
trigger messagecheck, review

**Comment by alebuh** [comment]
trigger ut8

**Pull request merged**